arch/mips/src/mips32/Toolchain.defs

```diff
@@ -245,14 +245,14 @@ endif
 # Optimization of unused sections
 
 ifeq ($(CONFIG_DEBUG_OPT_UNUSED_SECTIONS),y)
-  LDFLAGS          += --gc-sections
+  LDFLAGS          += -Wl,--gc-sections
```

Comment:
I think there was an automated way of adding -Wl in case if gcc is used for linking in case of ARM. Maybe the same approach can be used

Reply:
Yes, @rm5248 you can reference how arm support link with gcc and ld at the same time.

Reply:
It looks like the commit that I was basing this off of was reverted at some point, so I will configure this differently.

Reply:
Yes, you need check the last code.